src/cryptonote_core/service_node_list.cpp

```diff
@@ -95,40 +95,45 @@ namespace service_nodes
         }
 
         block_added_generic(block, txs);
-
-        if (real_height++ >= cache_state_from_height)
-          store_quorum_state_from_rewards_list(real_height);
       }
     }
 
     m_rollback_events.push_back(std::unique_ptr<rollback_event>(new prevent_rollback(current_height)));
   }
 
-  std::vector<crypto::public_key> service_node_list::get_service_nodes_pubkeys() const
+  std::vector<const cryptonote::account_public_address *> service_node_list::get_service_node_pubkeys() const
   {
-    std::vector<crypto::public_key> ret;
-    for (const auto& iter : m_service_nodes_keys)
-      ret.push_back(iter.second);
-    std::sort(ret.begin(), ret.end(),
-        [](const crypto::public_key& a, const crypto::public_key& b) {
-          return memcmp(reinterpret_cast<const void*>(&a), reinterpret_cast<const void*>(&b), sizeof(a)) < 0;
-        });
-    return ret;
+    std::vector<const cryptonote::account_public_address *> result;
```

Comment:
const crypto::public_key * or crypto::public_key

Using a pointer here instead of the whole public_key object saves 11ms on my machine. But it also makes it easier for people to make hard to find memory corruption bugs, because the pointers will be invalidated when the quorum for that height drops off the cache. For you this may be clear now, but for someone else using this function it may not be so obvious.

This is similar to returning the quorum_state, that the correct way to do this efficiently would be to use a shared pointer.

I have a personal preference for just doing the copy with crypto::public_key, and then only implementing microoptimizations if speed here actually becomes a problem.